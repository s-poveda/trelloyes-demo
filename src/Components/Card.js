import './card.css';
import React from 'react';
import { Component } from 'react';

class Card extends Component {
	static defaultProps= {title: '', content:''};

	render() {
		const { title, content } = this.props;
		return (
			<div className="Card">
    		<button type="button">delete</button>
    		<h3>{title}</h3>
    		<p>{content}</p>
    </div>
	)}
}

export default Card;
