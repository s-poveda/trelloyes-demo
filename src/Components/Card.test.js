import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';


describe('Card.js tests', () => {

		it('smoke test', () => {
				const div = document.createElement('div');
				ReactDOM.render( <Card />, div);
				ReactDOM.unmountComponentAtNode(div);
		});

		it('snapshot test without props', () => {
				const tree = renderer.create(<Card />).toJSON();
				expect(tree).toMatchSnapshot();
		});

		it('snapshot test with props', () => {
				const {title, content} = { title: 'First card', content: 'lorem ipsum' }
				const tree = renderer.create(<Card title={title} content={content}/>).toJSON();
				expect(tree).toMatchSnapshot();
		});


});
