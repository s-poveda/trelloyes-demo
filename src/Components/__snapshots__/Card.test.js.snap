// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.js tests snapshot test with props 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    First card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;

exports[`Card.js tests snapshot test without props 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    
  </h3>
  <p>
    
  </p>
</div>
`;
